desk/app/activity.hoon

```diff
@@ -382,45 +459,40 @@
       ~
     |=  [@ key=@da =event:a]
     ^-  [(unit event:a) ? @]
-    ?>  ?=(?(%post %reply %dm-post) -.event)
-    ?:  &(seen.event-parent =(time key))
-      [~ | ~]
-    ?.  =(-.event %reply)
-      [`event | ~]
-    ?:  (lth time.message-key.event reply-floor.event-parent)
+    ?:  &(seen =(time key))
       [~ | ~]
     [`event | ~]
   ==
 ++  stream-to-unreads
-  |=  =stream:a
+  |=  [=stream:a floor=time children=(list source:a)]
   ^-  unread-summary:a
-  =/  newest=(unit time)  ~
-  =/  count  0
-  =/  threads=(map message-id:a [oldest-unread=time count=@ud])  ~
+  =/  newest=time  floor
+  =/  total  0
+  =/  main  0
+  =/  notified=?  |
+  =/  main-notified=?  |
+  =|  last=(unit message-key:a)
   ::  for each event
   ::  update count and newest
   ::  if reply, update thread state
   |-
   ?~  stream
-    :+  (fall newest now.bowl)  count
-    %+  turn  ~(val by threads)
-    |=  [oldest-unread=time count=@ud]
-    [oldest-unread count]
+    [newest total notified ?~(last ~ `[u.last main main-notified]) children]
   =/  [[@ =event:a] rest=stream:a]  (pop:on-event:a stream)
-  =.  count  +(count)
-  =.  newest
-    ?>  ?=(?(%dm-post %post %reply) -.event)
-    ::REVIEW  should we take timestamp of parent post if reply??
-    ::        (in which case we would need to do (max newest time.mk.e))
-    `time.message-key.event
-  =?  threads  ?=(%reply -.event)
-    =/  old
-      %+  ~(gut by threads)  id.target.event
-      [oldest-unread=time.message-key.event count=0]
-    %+  ~(put by threads)  id.target.event
-    ::  we don't need to update the timestamp, because we always process the
-    ::  oldest message first
-    ::
-    [oldest-unread.old +(count.old)]
+  =/  volume  (get-volume -.event)
+  =?  notified  &(notify:volume notified.event)  &
+  ?.  ?&  unreads:volume
+          ?=(?(%dm-post %dm-reply %post %reply) -<.event)
+      ==
+    $(stream rest)
```

Comment:
```suggestion
  =?  notified  &(notify.volume notified.event)  &
  ?.  ?&  unreads.volume
          ?=(?(%dm-post %dm-reply %post %reply) -<.event)
      ==
    $(stream rest)
```

Also, this doesn't account for unreads on groups activity (joins, leaves) or other non-messaging activity. This is fine right now, because we don't try to display it anywhere, but this should probably get a `::TODO  support other event types`.